recipe/meta.yaml

```diff
@@ -23,8 +23,8 @@ build:
 # Need these up here for conda-smithy to handle them properly.
 requirements:
   build:
-    - {{ compiler('c') }}
-    - {{ compiler('cxx') }}
+    - clang =12.0.1
+    - clangxx =12.0.1
```

Comment:
AFAIU, we can't do it like that. We need the magic-jinja for the compiler to get the activation etc. right.

If you really need clang on linux (and I'm not sure we're ready for that, because we'd have to build against GCC's libstdcxx), then you should add something like to following to `conda_build_config.yaml` (and rerender)
```
c_compiler:                    # [linux]
  - clang                      # [linux]
c_compiler_version:            # [linux]
  - 14                         # [linux]
cxx_compiler:                  # [linux]
  - clangxx                    # [linux]
cxx_compiler_version:          # [linux]
  - 14                         # [linux]
```

CC @isuruf

Reply:
I will back out the change. It did not help, the same error persists.